templates/default/datadog.yaml.erb

```diff
@@ -112,6 +112,10 @@ if node['datadog']['statsd_forward_host']
   agent_config['statsd_forward_port'] = node['datadog']['statsd_forward_port']
 end
 
+if node['datadog']['agent6'] && node['datadog']['dogstatsd_socket']
+  agent_config['dogstatsd_socket'] = node['datadog']['dogstatsd_socket']
```

Comment:
if the default is left to `nil`, you can move set `dogstatsd_socket` directly in the block line 72 like the other regular config options. `nil` values are then automatically removed from the hash.